pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -487,6 +412,276 @@ def phase_str(phi: float) -> str:
             special_kwargs = dict(labelpad=10) if i == 0 else {}
             ax.set_ylabel(LABELS[i], fontsize=14, **special_kwargs)
 
+        if draw_target_regions:
+            target_regions = []  # [[start1, [targets1], end1],...]
+            for coords in ch_data.target:
+                targets = list(ch_data.target[coords])
+                tgt_strs = [str(q) for q in targets]
+                tgt_txt_y = max_amp * 1.1 - 0.25 * (len(targets) - 1)
+                tgt_str = "\n".join(tgt_strs)
+                if coords == "initial":
+                    x = t_min + final_t * 0.005
+                    target_regions.append([0, targets])
+                    if seqsamples._ch_objs[ch].addressing == "Global":
+                        axes[0].text(
+                            x,
+                            amp_top * 0.98,
+                            "GLOBAL",
+                            fontsize=13,
+                            rotation=90,
+                            ha="left",
+                            va="top",
+                            bbox=q_box,
+                        )
+                    else:
+                        axes[0].text(
+                            x,
+                            tgt_txt_y,
+                            tgt_str,
+                            fontsize=12,
+                            ha="left",
+                            bbox=q_box,
+                        )
+                else:
+                    ti, tf = np.array(coords) / time_scale
+                    target_regions[-1].append(ti)  # Closing previous regions
+                    target_regions.append(
+                        [tf + 1 / time_scale, targets]
+                    )  # New one
+                    for ax in axes:
+                        ax.axvspan(ti, tf, alpha=0.4, color="grey", hatch="//")
+                    axes[0].text(
+                        tf + final_t * 5e-3,
+                        tgt_txt_y,
+                        tgt_str,
+                        ha="left",
+                        fontsize=12,
+                        bbox=q_box,
+                    )
+
```

Comment:
what about `draw_phase_shifts` ? Are you not showing them because of `seq._basis_ref` ? I think you can get them from `ChannelSamples.phase`... (I have to dive a bit further in the code for this)

Reply:
ok